templates/registry/go/weaver.yaml

```diff
@@ -1,30 +1,58 @@
+params:
+  # you can specify namespaces here or pass them as a command line argument, it'd override the value here
+  # e.g. `weaver registry generate  --registry=../semantic-conventions/model --templates=./templates go ./output --param included_namespaces="db,http"`
+  # TODO -param as string slightly buggy - string inclusion is not the same as exact match with list element
+  included_namespaces: []
+
 templates:
+  # TODO - should we do it in the same way as here - https://github.com/open-telemetry/opentelemetry-go/blob/main/semconv/v1.36.0/httpconv/metric.go?
+  # i.e. generate attribute types inside the metric file
   - template: attr.go.j2
     application_mode: each
     file_name: "{{ctx.root_namespace}}/attributes.go"
+    # we're going to generate only the attributes that are used by the metrics we're generating
     filter: >
-      semconv_grouped_attributes
+      (semconv_grouped_metrics({"exclude_deprecated": true})
+      | [.[]
+        | select (($included_namespaces | length == 0) or (.root_namespace as $ns | $included_namespaces | index($ns)))
+        | .metrics?[]
+        | .attributes // []
+        | .[].name ]
+      | unique) as $relevant_attr_names
+      | semconv_grouped_attributes({"exclude_deprecated": true})
+      | map({root_namespace: .root_namespace, attributes: .attributes | map(select(.name as $n | $relevant_attr_names | index($n))) | select(length > 0)})
```

Comment:
Nice touch :) 

I appreciate the idea of generating less stuff and keeping our output cleaner